### PR TITLE
roomba code

### PR DESCRIPTION
**Pull request by dwaltsch**
/kipr/ sind die ganzen Header Files der rest ist min Roomba Code Pfusch

### CODE REVIEW

**Comment by dwaltsch** [comment]
@schmaenjael

File: roomba/main.c [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+    printf("%d,%d,%s",get_create_lflightbump(),0,"left light bumper \n");
+    printf("%d,%d,%s",get_create_lbump(),0,"left bumper \n");
+    printf("%d,%d,%s",get_create_llightbump(),0,"left light bumper \n");
+    printf("%d,%d,%s",get_create_lcliff(),0,"left cliff \n");
+    printf("%d,%d,%s",get_create_lwdrop(),0,"left wheel drop \n");
+    printf("%d,%d,%s",get_create_distance(),0,"total distance\n");
+    printf("%d,%d,%s",get_create_total_angle(),0,"total angle \n");
+}
+
+bool setupcamera(){
```
Comment (schmaenjael) [inline roomba/main.c:36]: A more concise way to write this:

```c
bool setupcamera(){
    if(camera_open() == 0 || camera_update() == 0) return false;
    return true;
}
```

File: roomba/main.c [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+    {
+        return false;
+    }
+    return true;
+}
+
+void setuprobot(bool debug, bool camera){
+    create_connect();
+    setupsensordata();
+    if(debug == true){
```
Comment (schmaenjael) [inline roomba/main.c:50]: Define Debug-Specific Code with C Preprocessor operators!

https://learn.microsoft.com/en-us/cpp/preprocessor/hash-if-hash-elif-hash-else-and-hash-endif-directives-c-cpp?view=msvc-170

File: roomba/main.c [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+        printsensordata();
+    }
+    if(camera){
```
Comment (schmaenjael) [inline roomba/main.c:53]: Opinionated: I like to reduce the depth of nesting -> If there is just one line of code that is after the ``if``, then remove the brackets and write it into the same line:

```c
  if (setupcamera()) printf("%s","Camera initialized");
  else printf("%s","Camera Error");
 ```

File: roomba/main.c [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+        camera_close();
+    }
+    disable_servos();
+    ao();
+    create_stop();
+    create_disconnect();
+}
+
+void drivestraight(int speed, bool direction, int distance, bool watchfortouch){ //false backwards true forwards distance in mm
```
Comment (schmaenjael) [inline roomba/main.c:74]: bool watchfortouch --> watch_for_touch

File: roomba/main.c [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+     while(get_create_total_angle() <= deg){
+         create_drive_direct(speed * -1,speed);
+     }
+    }
+    else{
+            while(get_create_total_angle() >= deg){
+                create_drive_direct(speed,speed * - 1);
+            }
+        }
+}
```
Comment (schmaenjael) [inline roomba/main.c:107]: Formatting!

File: roomba/main.c [hunk 6]
```diff
@@ -0,0 +1,173 @@
...
+
+void wasd(){
+    bool exit;
+    exit = true;
+    int speed = 100;
+    while(exit){
+        char input;
+        printf("%s","Please provide input:\n");
+        scanf("%c",&input);
+        if(input == 'w'){
```
Comment (schmaenjael) [inline roomba/main.c:117]: This could be solved cleaner by using a ``switch`` statement!

File: roomba/main.c [hunk 7]
```diff
@@ -0,0 +1,173 @@
...
+            exit = false;
+        }
+        if(input == 'm'){
+            printf("%s","Please provide speed:\n");
+            scanf("%d",&speed);
+        }
+    }
+}
+
+void grapscher(int startposition, int endposition, int port){ // opens and closes claw
```
Comment (schmaenjael) [inline roomba/main.c:139]: Function name is ambiguous, rename ``grapscher`` to a more reasonable name! 

``startposition`` and ``endposition`` do not fulfill the C naming standard -> ``start_position`` and ``end_position``!

File: roomba/main.c [hunk 8]
```diff
@@ -0,0 +1,173 @@
...
+}
+
+void movearm(int position, bool grabobject,int rotation, int lowerservoid, int grabservoid, int speed){
+    int idlegrabposition = 500; // placeholder
+    if(rotation != 0){ // turn
+        turn(speed, rotation);
+    }
+    moveservo(lowerservoid,position);
+    // claw
+    if(grabobject == true){
```
Comment (schmaenjael) [inline roomba/main.c:153]: boolean comparisions to ``true`` are not necessary! Remove the comparison!

File: roomba/main.c [hunk 9]
```diff
@@ -0,0 +1,173 @@
...
+        grapscher(idlegrabposition,550,grabservoid); //placeholder value
+    }
+}
+
+int getObject(int channel){
```
Comment (schmaenjael) [inline roomba/main.c:158]: This function does not return an ``integer``!

Opinionated: You can reduce unnecessary nesting by inverting the ``if-statement``:

```c
void getObject(int channel){
    if(!get_object_count(channel)) return;
    camera_update();
    
    get_object_center_x(channel, 0);
    get_object_center_y(channel, 0);
}
```

File: roomba/main.c [hunk 10]
```diff
@@ -0,0 +1,173 @@
...
+    }
+}
+int main()
+{
+    setuprobot(false,false);
+    wasd();
+    turn(100,90);
+    shutdownrobot(false);
+    return 0;
+}
```
Comment (schmaenjael) [inline roomba/main.c:173]: Please add a new-line at the end of the file 😃

File: roomba/main.c (hunk 7 above)
Comment (dwaltsch) [reply roomba/main.c:139]: already removed

**Pull request merged**